integration_tests/src/main/python/arithmetic_ops_test.py

```diff
@@ -500,3 +501,32 @@ def test_greatest(data_gen):
             lambda spark : gen_df(spark, gen).select(
                 f.greatest(*command_args)), conf=allow_negative_scale_of_decimal_conf)
 
+def _assert_div_by_zero(session_func, test_div_func, conf):
+    with pytest.raises(Py4JJavaError) as py4jError:
+        session_func(test_div_func, conf)
+    assert 'java.lang.ArithmeticException: divide by zero' in str(py4jError.value.java_exception)
+
+@pytest.mark.parametrize('ansi_mode', ['nonAnsi', 'ansi'])
+@pytest.mark.parametrize('exp_type', ['bothLiterals', 'justZeroLiteral', 'noLiterals'])
+def test_div_by_zero(ansi_mode, exp_type):
+    if ansi_mode == 'ansi':
+        if is_before_spark_310() or exp_type != 'bothLiterals':
+            pytest.xfail('https://github.com/apache/spark/pull/29882')
+
+    ansi_conf = {'spark.sql.ansi.enabled': ansi_mode == 'ansi'}
+    data_gen = lambda spark: two_col_df(spark, IntegerGen(), IntegerGen(min_val=0, max_val=0), length=1)
+
+    if exp_type == 'bothLiterals':
+        div_by_zero_func = lambda spark: data_gen(spark).select(f.lit(1) / f.lit(0))
+    elif exp_type == 'justZeroLiteral':
+        div_by_zero_func = lambda spark: data_gen(spark).select(f.col('a') / f.lit(0))
+    else:
+        div_by_zero_func = lambda spark: data_gen(spark).select(f.col('a') / f.col('b'))
+
+    if ansi_mode == 'ansi':
+        div_by_zero_func_collect = lambda spark: div_by_zero_func(spark).collect()
+        _assert_div_by_zero(with_cpu_session, div_by_zero_func_collect, ansi_conf)
```

Comment:
I'm not sure we need to test that the CPU version does what we expect it to do. If we had something that would adjust our behavior to match what the CPU does, that might be interesting, but if we are not doing that, then I would just want to check that we do what we expect.

Reply:
this is in line how we assert getting equivalent results for non exception cases via `assert_gpu_and_cpu_are_equal_collect`. Per your suggestion, I can move it to the generic layer and have something like `assert_gpu_and_cpu_same_error(func, conf, errorMsg)`